src/main/java/org/alfresco/repo/security/authentication/AuthenticationServiceImpl.java

```diff
@@ -186,10 +195,15 @@ public void recordFailedAuthentication(String userName)
                     }
                 }
             }
-            protectedUsersCache.put(userName, protectedUser);
+            protectedUsersCache.put(protectedUserKey, protectedUser);
         }
     }
 
+    private String getProtectedUserKey(String userName)
```

Comment:
Javadoc to say what the method does.  It's private so there is no way to guess that it combines the service instance ID with the username.

Reply:
I will make it public and add the java doc.

Reply:
Private is fine.  But it needs to be commented.